src/gc/gc.cpp

```diff
@@ -308,72 +308,55 @@ void GCStatistics::DisplayAndUpdate()
 
 #endif // GC_STATS
 
-#ifdef BIT64
-#define TOTAL_TIMES_TO_SHIFT 6
-#else
-#define TOTAL_TIMES_TO_SHIFT 5
-#endif // BIT64
-
 size_t round_up_power2 (size_t size)
 {
-    unsigned short shift = 1;
-    size_t shifted = 0;
-
-    size--;
-    for (unsigned short i = 0; i < TOTAL_TIMES_TO_SHIFT; i++)
-    {
-        shifted = size | (size >> shift);
-        if (shifted == size)
-        {
-            break;
-        }
-
-        size = shifted;
-        shift <<= 1;
-    }
-    shifted++;
-
-    return shifted;
+    // Get the 0-based index of the most-significant bit in the size.
+    // If the call failed (because `size` is zero), return 0.
+    DWORD msb_index;
+    if (0 ==
+#ifdef BIT64
+        BitScanReverse64(
+#else
+        BitScanReverse(
+#endif
+            &msb_index, size)) { return 0; }
+
+    // If `size` is already a power-of-two value, return it as-is.
+    // Otherwise, round it up to the next power-of-two by shifting
+    // it's MSB upwards one position.
+    size_t msb_mask = static_cast<size_t>(1) << msb_index;
+    // Always do this so the compiler uses e.g. cmov (x86) or csel (arm) below instead of a branch.
+    size_t rounded_up = msb_mask << 1;
+    return
+        msb_mask == size
+        ? size
+        : rounded_up;
 }
 
 inline
 size_t round_down_power2 (size_t size)
 {
     size_t power2 = round_up_power2 (size);
+    size_t rounded = power2 >> 1;
 
-    if (power2 != size)
-    {
-        power2 >>= 1;
-    }
-
-    return power2;
+    return power2 == size ? size : rounded;
 }
 
-// the index starts from 0.
+// Get the 0-based index of the most-significant bit in the value.
+// Returns -1 if the input value is zero (i.e. has no set bits).
+inline
 int index_of_set_bit (size_t power2)
```

Comment:
If you are changing the semantics of this function, you should also change its name, e.g., `index_of_msb`.

Reply:
And the parameter name `power2` is misleading now.

Reply:
I don't think the semantics of this function have changed. The old version was just implemented as a binary search over the bits of ``power2`` rather than using a compiler intrinsic. ``power2`` wasn't required to be a power-of-two value before either, but I agree the parameter name is misleading -- I'll change it.

Reply:
// `power2` wasn't required to be a power-of-two value before either //
That is not true.  The old version guaranteed to return the index of the msb for powers of two only.  For other values the `if (power2 & temp) { return mid; }` branch might return an index of some other bit set to 1.  For instance, consider the `0b1...1` value passed to the original function.

Reply:
@jack-pappas Do you agree with renaming the function?

Reply:
Yes, I agree with renaming the function -- my mistake, I thought I had done that as part of my last update to the PR. I've made the change now though (and updated the PR).